languages/python/oso/oso/enforcer.py

```diff
@@ -122,3 +122,62 @@ def authorized_actions(self, actor, resource, allow_wildcard=False) -> list:
             actions.add(action)
 
         return list(actions)
+
+    def authorize_field(self, actor, action, resource, field):
+        """Ensure that ``actor`` is allowed to perform ``action`` on a given
+        ``resource``'s ``field``.
+
+        If the action is permitted with an ``allow_field`` rule in the policy,
+        then this method returns ``None``. If the action is not permitted by the
+        policy, this method will raise a ``ForbiddenError`.
+
+        :param actor: The actor performing the request.
+        :param action: The action the actor is attempting to perform on the
+        field.
+        :param resource: The resource being accessed.
+        :param field: The name of the field being accessed.
+        """
+        if not self.policy.query_rule_once(
+            "allow_field", actor, action, resource, field
+        ):
+            raise self._get_error(False)
+
+    def authorized_fields(self, actor, action, resource, allow_wildcard=False) -> list:
+        """Determine the fields of ``resource`` on which ``actor`` is allowed to
```

Comment:
Pretty sure this type hint should be `List[str]` since we support Python < 3.9. https://mypy.readthedocs.io/en/stable/builtin_types.html?highlight=list#generic-types

If CI isn't catching this we should look into why.

Reply:
It can't be `List[str]` though because you might get an enum type back as an action. `List[any]`?